torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -147,23 +149,24 @@ def __init__(
         parameters_as_bucket_view: bool = False,
         **default: Any,
     ):
+        # Perform type and assumption checks on the input parameters
+        self._verify_and_init_params(params)
+        self._verify_same_param_device()
+        self._verify_same_dense_param_type()
+        self._device = self._all_params[0].device
+
         # Hold all the model params in the root .param_groups
         # NOTE: the default constructor uses `add_param_group` which is partially overloaded here
         # we introduce the `initialized` flag for be able to dissociate the behaviour of
         # `add_param_group` in between super() and ZeroRedundancyOptimizer
         self.initialized = False
-        super().__init__(params, default)
+        super().__init__(self._all_params, default)
 
-        # Partition information. lazy evaluation, computed if requested
-        self._per_device_params_cache: "OrderedDict[torch.device, List[List[Parameter]]]" = (
-            OrderedDict()
-        )  # device, rank, params
+        # Partition information (evaluated lazily)
         self._param_rank_cache: Dict[torch.Tensor, int] = {}
```

Comment:
This is prior to this PR. I wonder if the name should be "_param_to_rank_cache" to stay consistent with other names?

Reply:
Makes sense. Renamed to `_param_to_rank_cache`.